### PR TITLE
Initial code migration

### PR DESCRIPTION
**Pull request by SuperQ**
Migrate code from https://github.com/prometheus/common/sigv4.

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* Julien Pivotto <roidelapluie@prometheus.io> @roidelapluie
+* Josue (Josh) Abreu <josue.abreu@gmail.com> @gotjosh
+* Arthur Sens <arthur@prometheus.io> @ArthurSens
```
Comment (ArthurSens) [inline MAINTAINERS.md:3]: I'd love to not be a maintainer of this part if that's ok 😅

**Comment by SuperQ** [comment]
Ping, I'd like to get the CI up and the deprecation process started.

**Review by ArthurSens — APPROVED** [review]
LGTM, but it would be nice to get approvals from the people listed as maintainers

**Comment by SuperQ** [comment]
@roidelapluie approved the the transfer https://github.com/prometheus/common/issues/709.

**Pull request merged**